crates/engine/tree/src/tree/root.rs

```diff
@@ -971,20 +989,25 @@ where
 }
 
 fn extend_multi_proof_targets(targets: &mut MultiProofTargets, other: MultiProofTargets) {
-    for (address, slots) in other {
-        targets.entry(address).or_default().extend(slots);
+    targets.accounts.extend(other.accounts);
+
+    for (address, slots) in other.storages {
+        targets.storages.entry(address).or_default().extend(slots);
     }
 }
 
 fn extend_multi_proof_targets_ref(targets: &mut MultiProofTargets, other: &MultiProofTargets) {
-    for (address, slots) in other {
-        targets.entry(*address).or_default().extend(slots);
+    targets.accounts.extend(&other.accounts);
+
+    for (address, slots) in &other.storages {
+        targets.storages.entry(*address).or_default().extend(slots);
     }
```

Comment:
we have this twice, should these be a helper function on MultiProofTargets instead?